tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -37,6 +40,25 @@ function create_runtimeclasses() {
 	echo "Creating the runtime classes"
 
 	for shim in "${shims[@]}"; do
+		if [[ "${USING_NFD}" == "true" ]]; then
+
+			# Safety check to ensure NFD worker is deployed and is running
+			[[ $(kubectl get pods -n node-feature-discovery | grep nfd-worker) ]] || die "NFD worker was not found in the cluster"
+			[[ $(kubectl get pods -n node-feature-discovery | grep nfd-worker | grep Running) ]] || die "NFD worker is not currently running"
+
+			if [[ " ${!tees_relying_on_nfd[*]} " =~ " ${shim} " ]]; then
+				tee=${tees_relying_on_nfd[${shim}]}
+				echo "As the cluster relies on NodeFeatureDiscovery, let's create the NodeFatureRule for ${shim}"
+				kubectl apply -f /opt/kata-artifacts/nodefeaturerules/${tee}-nfd-rule.yaml
+
+				echo "Uncomment the value from the kata-${shim} podOverhead definition"
+				sed -i -e 's|#'${tee}'|'${tee}'|' /opt/kata-artifacts/runtimeclasses/kata-${shim}.yaml
+
+				echo "NodeFeatureRule created for ${shim}"
+				cat /opt/kata-artifacts/runtimeclasses/kata-${shim}.yaml
```

Comment:
> this might be problematic though if the overhead resource is not known to `kata-deploy.sh`.

one question occurred to me: `kata-deploy.sh` is a per-node script isn't it? We will have to be careful creating cluster-wide resources from each node.

Reply:
also in case of NFD, it would be good to use the TEE labels as a filter to `runtimeclass.scheduling.nodeSelector`.